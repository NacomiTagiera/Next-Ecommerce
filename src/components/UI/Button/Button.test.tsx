import { render, screen } from "@testing-library/react";
import { userEvent } from "@testing-library/user-event";

import { Button } from ".";

describe("Button", () => {
	it("renders with default props", () => {
		const { asFragment } = render(<Button />);

		expect(asFragment()).toMatchSnapshot();
	});

	it("displays its content", () => {
		render(
			<Button>
				Click <b>Me</b>
			</Button>,
		);

		expect(screen.getByText(/Click/)).toBeVisible();
		expect(screen.getByText(/Me/)).toBeVisible();
	});

	it("assigns class names by properties", () => {
		render(<Button variant="outlined" size="lg" />);

		expect(screen.getByRole("button")).toHaveClass(
			"bg-transparent border border-slate-200 hover:bg-slate-100 text-base px-5 py-3",
		);
	});

	it("injects native button properties", () => {
		render(<Button title="Click me">Click me</Button>);

		expect(screen.getByTitle("Click me")).toBeVisible();
	});

	it("renders a link when href is provided", () => {
		render(<Button href="/about">About</Button>);

		expect(screen.getByRole("link")).toHaveAttribute("href", "/about");
	});

	it("renders leading and trailing icons", () => {
		const leadingIcon = () => <svg data-testid="leading" />;
		const trailingIcon = () => <svg data-testid="trailing" />;
		render(
			<Button leadingIcon={leadingIcon} trailingIcon={trailingIcon}>
				Click me
			</Button>,
		);

		expect(screen.getByTestId("leading")).toBeVisible();
		expect(screen.getByTestId("trailing")).toBeVisible();
	});

	it("handles correctly disabled state", async () => {
		const onClick = jest.fn();
		render(
			<Button aria-disabled onClick={onClick}>
				Click me
			</Button>,
		);

		await userEvent.click(screen.getByRole("button"));
		expect(onClick).not.toHaveBeenCalled();
	});
});
