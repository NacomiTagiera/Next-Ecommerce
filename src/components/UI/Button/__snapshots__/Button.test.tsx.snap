// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders with default props 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center font-medium shadow-sm focus:ring-4 focus:outline-none rounded-md text-center transition-colors aria-disabled:cursor-not-allowed aria-disabled:shadow-none aria-disabled:pointer-events-none text-white bg-skyfall-500 hover:bg-skyfall-600 focus:ring-skyfall-300 aria-disabled:bg-zinc-500 text-sm px-5 py-2.5"
  />
</DocumentFragment>
`;
